### DISCUSSION TITLE
How to enable context (history) for gpt 3.5 turbo or gpt 4?

### DISCUSSION THREAD

**Discussion by TheGreatApollyon** [xtekky/gpt4free#1082] [Q&A]
How can I enable context history, regardless of whether it does it itself or I'll have to provide a json of previous queries and responses, so that the queries made henceforth are influenced by previous context.

What I'm trying to achieve:
User: (Message 1) Tell me about India.
Response 1: _________________
User: (Message 2) Where is it located?
Response 2: Southern Asia

**Comment by itschasa** [comment]
did it work via mentioning the whole conversation in the payload of messages?

**Reply by TheGreatApollyon** [reply]
I presume there are two problems with that:
1) Prompt length limit
2) Longer processing time with every new message

It'll work for sometime appropriately, what I think might be optimal is to keep only last n (5~10) conversations in memory and pass it as context for now
